### ISSUE TITLE
C++11 nullptr

### ISSUE THREAD

**Issue by tbenthompson** [onqtam/doctest#27]
Howdy,

I'm converting a test setup using Catch to doctest. Most of the process was super easy -- just a global find and replace for catch --> doctest and SECTION -> SUBCASE. But, I'm getting a couple errors.

1) I'm using a check like:

```
REQUIRE(ptr == nullptr);
```

and getting an error like:

```
/home/tbent/projects/taskloaf/taskloaf/lib/doctest.h:267:63: error: ambiguous overload for ‘operator<<’ (operand types are ‘std::ostream {aka std::basic_ostream<char>}’ and ‘std::nullptr_t’)
             static const bool value = sizeof(testStreamable(s << t)) == sizeof(yes);
````
It looks like this was an issue in Catch previously: https://github.com/philsquared/Catch/issues/80
and for me it's solved by adding this:
```
namespace doctest {
template <>
struct StringMaker<std::nullptr_t> {
    static String convert(std::nullptr_t) { return "nullptr"; }
};
}
```

2) Comparing the size of a collection like:
```
REQUIRE(my_vector.size() == 4);
```
produces
```
comparison between signed and unsigned integer expressions
```
warnings. While, this is less of a problem than #1, it's always nice to get rid of warnings!

Cheers!

**Comment by onqtam** [comment]
thanks for the reports! the 2nd is reported in issue #16

both things will be fixed for 1.1 - hopefully in 1 or 2 weeks.

**Comment by tbenthompson** [comment]
Awesome. Thanks.
